### PR TITLE
PAYARA-2425 Fix for using multiple envrionment variable in a line

### PR DESCRIPTION
**Pull request by Cousjava**
* PAYARA-2425 Fix for using multiple envrionment variable in a line

* PAYARA-2425 Add - to start of option if not already present

* PAYARA-2425 Fix for create-jvm-options using an envrionment variable

* PAYARA-2425 added comment

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins, quick test Payara 4 please

**Comment by Cousjava** [comment]
Jenkins, quick test Payara 4 please

File: nucleus/admin/launcher/src/main/java/com/sun/enterprise/admin/launcher/JvmOptions.java [hunk 1]
```diff
@@ -58,9 +58,19 @@
...
 class JvmOptions {
 
+    /**
+     * Pattern for recognising environment variables which are in the style of ${ENV=foo}
+     * @see org.glassfish.config.support.TranslatedConfigView
+     */
     private final static Pattern envP = Pattern.compile("([^\\$]*)\\$\\{ENV=([^\\}]*)\\}([^\\$]*)");
     private static final int MAX_SUBSTITUTION_DEPTH = 100;
     
+    Map<String, String> sysProps = new HashMap<String, String>();
```
Comment (arjantijms) [inline nucleus/admin/launcher/src/main/java/com/sun/enterprise/admin/launcher/JvmOptions.java]: You don't need the extra generic parameters here:

    Map<String, String> sysProps = new HashMap<();

File: nucleus/admin/launcher/src/main/java/com/sun/enterprise/admin/launcher/JvmOptions.java [hunk 2]
```diff
@@ -80,11 +90,14 @@
                 String matchValue = m2.group(2).trim();
                 String newValue = System.getenv(matchValue);
                 if (newValue != null) {
-                    s = newValue;
+                   s = m2.replaceFirst(Matcher.quoteReplacement(m2.group(1) + newValue + m2.group(3)));
```
Comment (arjantijms) [inline nucleus/admin/launcher/src/main/java/com/sun/enterprise/admin/launcher/JvmOptions.java]: Avoid names like `s' and `m2`

**Comment by arjantijms** [comment]
Jenkins, quick test Payara 4 please

**Pull request merged**